### ISSUE TITLE
Inner hits source filtering not working

### ISSUE THREAD

**Issue by mattweber** [elastic/elasticsearch#23090]
When using nested inner hits feature and source filtering on the inner hit the included fields are not returned.  This is specifying the include path relative to the nested object (which is how it is returned in results without filtering).  If I specify the include using the absolute path, I get a result but the source is no longer returned relative to the nested object.

If I have this document:

```json
{"nested":[{"nested_field": "some value"}]}
```

Without filtering, a nested inner hit response would return:

```json
..."_source": {"nested_field": "some value"}
```

If I set the inner hit source filtering to include `nested_field` I get:

```json
..."_source": {}
```

And if I set the inner hit source filtering to include `nested.nested_field` I get:

```json
..."_source": {"nested": {"nested_field": "some value"}}
```

This is on elasticsearch 5.2.0.

/cc @martijnvg @tlrx

**Comment by clintongormley** [comment]
Bugger... the behaviour of source filtering was changed in https://github.com/elastic/elasticsearch/pull/18567 but without the corresponding change to unfiltered source.  Fixing this now would be a breaking change, either way.  Perhaps we should aim to fix this in 6.0 instead.

**Comment by mattweber** [comment]
Waiting for 6.0 is fine with me, its not a pressing issue.  It actually only popped up in some tests I have for a clients internal project.  Thanks.

**Comment by mattweber** [comment]
Just hit this bug again and reminded to check on its status.  @martijnvg @clintongormley has anything been done on this for the 6x release?

**Comment by martijnvg** [comment]
@mattweber I think this bug fell through the cracks. I'll look into it right away.

**Comment by sand33p-23** [comment]
Is the fix only available in 6.x release or anything has been done for 5.x versions?

**Comment by martijnvg** [comment]
@sand33p-23 The fix is available from 6.0. it has not been backported to a 5.x version.
